src/module-elasticsuite-catalog/Model/Product/Indexer/Fulltext/Datasource/PriceData.php

```diff
@@ -65,10 +76,26 @@ public function addData($storeId, array $indexData)
             $originalPrice = $priceModifier->getOriginalPrice($priceDataRow);
             $price         = $priceModifier->getPrice($priceDataRow);
 
+            $isDiscount    = $price < $originalPrice;
+            if (in_array($productTypeId, $this->attributeResourceModel->getCompositeTypes())) {
+                $isDiscount = false;
+                $priceModifier = $this->getPriceDataReader('default');
+                $allChildrenIds = $this->attributeResourceModel->loadChildrens([$productId], $storeId);
```

Comment:
This causes a load query per each configurable product in the batch.

It's possible to load all of them at once, like in https://github.com/Smile-SA/elasticsuite/blob/2.10.x/src/module-elasticsuite-catalog/Model/Product/Indexer/Fulltext/Datasource/AttributeData.php#L70

Maybe even better : at this step, we might already have the "children_ids" key inside the product data (that's true if the AttributeData datasource is run before the Price, which I'm not sure about) : 

https://github.com/Smile-SA/elasticsuite/blob/2.10.x/src/module-elasticsuite-catalog/Model/Product/Indexer/Fulltext/Datasource/AttributeData.php#L87